### ISSUE TITLE
Not working with modpacks [FORGE]

### ISSUE THREAD

**Issue by ChaoticWagon** [sp614x/optifine#5570]
## Description of Issue

As soon as I add optifine to most of my curseforge modpacks, it crashes when the mods are finished loading, or the mod loader is on the "Finishing up" or "Freezing Data" Screen. ModPacks work without optifine.

## Steps to Reproduce

1. Add the correct version of optifine to the mods folder in the desired modpack.
2. start the modpack

## OptiFine Version

Depends on the modpack... Bug has been found on (1.16.5, 1.12.2, and 1.15.2)

## Installation Method
dropped into mods folder

## Fabric/Forge Version
Forge Versions forge-14.23.5.2854 and forge-36.1.4

## Other Installed Mods

Theres a lot of mods so ill just say the modpacks I'm using. Valhelsia 3 and SevTech Ages

## Log Files/Crash Reports
Doesnt return a crash report or log file, only says error 0

## F3 Debug Screenshot

Cant even open game.

## Prior Testing

works fine without optifine

**Comment by KaiAF** [comment]
Can you find what mod is causing this? 
You can try doing this:
- Split the mods folder into two groups
- repeat until you find what mod is causing the issue.
- Then tell us what mod is causing the issue.

**Comment by ChaoticWagon** [comment]
Ill do my best but i will take a while because these modpacks have 100+ mods

**Comment by ChaoticWagon** [comment]
Here is the Mod list for Valhelsia 3:
Waddles (by Girafi)
RecipeBuffers (by malte0811)
Just Enough Resources (JER) (by way2muchnoise)
Music Maker Mod (by xerca0)
Inventory Sorter (by cpw)
Create (by simibubi)
Polymorph (Forge) (by TheIllusiveC4)
Macaw's Roofs (by sketch_macaw)
Netherite Horse Armor Mod (by louie_cai)
Customizable Elytra (by Hidoni)
Tool Stats (by DarkhaxDev)
In Control! (by McJty)
Cosmetic Armor Reworked (by LainMI)
PackMenu (by Shadows_of_Fire)
Structure Gel API (by ModdingLegacy)
Health Overlay (by Terrails)
Absent by Design (by Lothrazar)
Botania (by Vazkii)
Bayou Blues (by ex0planetary)
BlockCarpentry (by PianoManu)
Abundance (by ex0planetary)
Step (by ModdingLegacy)
Tetra (by mickelus)
Bedspreads (Forge) (by TheIllusiveC4)
Powah! (by owmii)
FTB GUI Library (by FTB)
Snow! Real Magic! ⛄ (by Snownee_)
Mekanism Additions (by bradyaidanc)
Mekanism (by bradyaidanc)
spark (by Iucko)
Open Loader (by DarkhaxDev)
Mekanism Generators (by bradyaidanc)
Buzzier Bees (by TeamAbnormals)
Kiwi 🥝 (by Snownee_)
Mekanism Tools (by bradyaidanc)
CoFH Core (by TeamCoFH)
Ender Storage 1.8.+ (by covers1624)
Simple Discord Rich Presence (by Sunekaer)
Better Biome Blend (by FionaTheMortal)
Druidcraft (by VulpTheHorseDog)
Decorative Blocks modded compat (by Evo_934)
Abnormals Core (by TeamAbnormals)
Better Badlands (by Gkoliver)
Waystones (by BlayTheNinth)
KubeJS Immersive Engineering (by LatvianModder)
Comforts (Forge) (by TheIllusiveC4)
Placebo (by Shadows_of_Fire)
MythicBotany (by noeppinoeppi)
iChunUtil (by ohaiiChun)
JAOPCA (by TheLMiffy1111)
Back Tools (by ohaiiChun)
Better Advancements (by way2muchnoise)
The Conjurer (by ModdingLegacy)
Fishing Real (by PricelessKoala)
Flux Networks (by sonar_sonic)
Bookshelf (by DarkhaxDev)
KleeSlabs (by BlayTheNinth)
CC: Tweaked (by SquidDev)
AIOT Botania (by MelanX)
Curios Quark Oddities Backpack (by 6LeoMC)
Forbidden and Arcanus (by cesar_zorak)
KubeJS UI (by LatvianModder)
Eidolon (by elucent_)
Lazy DataFixerUpper(LazyDFU) [FORGE] (by Corgi_Taco)
Game Stages (by DarkhaxDev)
OpenBlocks Elevator (by vsngarcia)
Savage & Ravage (by TeamAbnormals)
Structurize (by H3lay)
Waila Harvestability (by squeek502)
AI Improvements (by QueenOfMissiles)
The Endergetic Expansion (by TeamAbnormals)
FTB Backups (by FTB)
Client Tweaks (by BlayTheNinth)
YUNG's API (Forge) (by YUNGNICKYOUNG)
Allurement (by TeamAbnormals)
Light Overlay (Rift/Forge/Fabric) (by shedaniel)
Sound Device Options (by ichttt)
Environmental (by TeamAbnormals)
Ensorcellation (by TeamCoFH)
Autumnity (by TeamAbnormals)
Rhino (by LatvianModder)
Carry On (by Tschipp)
Cherished Worlds (Forge) (by TheIllusiveC4)
Berry Good (by TeamAbnormals)
Refined Storage (by raoulvdberge)
Armor Toughness Bar (by tfarecnim)
Pane In The Glass (by MongoTheElder)
Curious Elytra (Forge) (by TheIllusiveC4)
Chocolate Fix (by AlcatrazEscapee)
JEI Integration (by SnowShock35)
Curios API (Forge) (by TheIllusiveC4)
Immersive Engineering (by BluSunrize)
ConnectedTexturesMod (by tterrag1098)
Peculiars (by cosmicgelatin1)
Crafting Tweaks (by BlayTheNinth)
Snow Under Trees (by bl4ckscor3)
MineColonies for ComputerCraft (by uecasm)
KubeJS (by LatvianModder)
YUNG's Better Caves (Forge) (by YUNGNICKYOUNG)
Macaw's Bridges (by sketch_macaw)
Archer's Paradox (by TeamCoFH)
Farmer's Delight (by vectorwing)
Transport (by Sky_Som)
Quark (by Vazkii)
Industrial Foregoing (by Buuz135)
AutoRegLib (by Vazkii)
Quark Oddities (by Vazkii)
Engineer's Decor (by wilechaote)
Macaw's Furniture (by sketch_macaw)
Torchmaster (by xalcon)
Architectury API (Forge) (by shedaniel)
Paintings ++ (by AbsolemJackdaw)
Scuba Gear (by ModdingLegacy)
Torch Slabs Mod (by EndlesNights)
Xaero's World Map (by xaero96)
AmbientSounds 3 (by CreativeMD)
Extended Lights (by Polyvalord)
Better Burning (by DarkhaxDev)
Valhelsia Core (by ValhelsiaTeam)
The Mighty Architect (by simibubi)
Biomes O' Plenty (by Forstride)
Controlling (by Jaredlll08)
Better Title Screen (by Girafi)
Blood Magic (by WayofTime)
YUNG's Better Mineshafts (Forge) (by YUNGNICKYOUNG)
Refined Storage: Requestify (by Buuz135)
Cooking for Blockheads (by BlayTheNinth)
Immersive Petroleum (by Flaxbeard)
SwingThroughGrass (by exidex)
Tetranomicon (by syrikalis)
Atmospheric (by TeamAbnormals)
KubeJS Mekanism (by LatvianModder)
FerriteCore (Forge) (by malte0811)
LibX (by noeppinoeppi)
Cat Jammies (by Buuz135)
AppleSkin (by squeek502)
Dash (by ModdingLegacy)
Fast Leaf Decay (by olafskiii)
Apotheosis (by Shadows_of_Fire)
Just Enough Items (JEI) (by mezz)
Iron Chests (by ProgWML6)
FastWorkbench Minus Replacement (by tfarecnim)
Lollipop (by owmii)
Cloth Config API (Forge) (by shedaniel)
Mowzie's Mobs (by bobmowzie)
Macaw's Windows (by sketch_macaw)
Overloaded Armor Bar (by tfarecnim)
Caelus API (Forge) (by TheIllusiveC4)
Extra Boats (by TeamAbnormals)
ReAuth (by TechnicianLP)
Default Options (by BlayTheNinth)
Patchouli (by Vazkii)
Notes (by Chaosyr)
MineColonies (by H3lay)
RandomPatches (Forge) (by TheRandomLabs)
Aquaculture 2 (by Shadowclaimer)
FastFurnace (by Shadows_of_Fire)
Artifacts (by ochotonida)
Neon Craft Mod (by Polyvalord)
Save My Stronghold! (Forge) (by YUNGNICKYOUNG)
Leap (by ModdingLegacy)
Macaw's Doors (by sketch_macaw)
ServerConfig Updater (by Darkere)
mGui (by mickelus)
Performant (by someaddon)
Macaw's Trapdoors (by sketch_macaw)
Personality (by TeamAbnormals)
Alex's Mobs (by alex1the1666)
ObserverLib (by HellFirePvP)
Dark Utilities (by DarkhaxDev)
Supplementaries (by MehVahdJukaar)
Abnormals Delight (by TeamAbnormals)
Runelic (by DarkhaxDev)
Neapolitan (by TeamAbnormals)
Valhelsia Structures (by ValhelsiaTeam)
Server Tab Info (by black_dog20)
Windowlogging (by mc_Grimmauld)
Decorative Blocks (by stohun)
CreativeCore (by CreativeMD)
Miner's Helmet (by ModdingLegacy)
KubeJS Create (by LatvianModder)
Toast Control (by Shadows_of_Fire)
Psi (by Vazkii)
Akashic Tome (by Vazkii)
Refined Storage Addons (by raoulvdberge)
Titanium (by hrznstudio)
Mouse Tweaks (by YaLTeR)
Upgrade Aquatic (by TeamAbnormals)
Dynamic Surroundings (by OreCruncher)
Storage Drawers (by Texelsaur)
Clumps (by Jaredlll08)
Curio of Undying (Forge) (by TheIllusiveC4)
Immersive Posts (by TwistedGate)
Chisels & Bits (by AlgorithmX2)
Updated Engineering (by tdstress)
KubeJS Blood Magic (by LatvianModder)
Inventory HUD+ (by dmitrylovin)
Curious Shulker Boxes (Forge) (by TheIllusiveC4)
Storage for ComputerCraft (by uecasm)
Enchanted Book Redesign (by tfarecnim)
Enhanced Mushrooms (by ex0planetary)
Xaero's Minimap (by xaero96)
Fairy Lights (by pau101)
Bed Benefits (by DarkhaxDev)
Citadel (by alex1the1666)
CodeChicken Lib 1.8.+ (by covers1624)
TrashSlot (by BlayTheNinth)
Nature's Compass (by Chaosyr)
Curious Armor Stands (by ochotonida)
Neat (by Vazkii)
Valhelsia Tweaks (by ValhelsiaTeam)
Goblin Traders (by MrCrayfish)
Astral Sorcery (by HellFirePvP)
Mystical World (by EpicSquid315)
Jade 🔍 (by Snownee_)

The first time I used optifine on this pack it actually opened fine, but it crashed immediately when I tried turning on Sildurs Viberant Medium shaders. I remember it said it had trouble loading Immersive engineering. IDK if that helps but it wouldn't open anymore after that with optifine.
